.github/workflows/main.yml

```diff
@@ -201,8 +209,15 @@ jobs:
           cd docs/docsgen && make text
         continue-on-error: true
 
-      - name: Run Python tests with numpy<2.0
-        if: matrix.python_version == '3.11' || matrix.python_version == '3.12'
+      - name: Run Python tests with numpy<2.0 (win, mac)
+        if: (matrix.python_version == '3.11' || matrix.python_version == '3.12') &&  (matrix.os == 'windows-latest' || matrix.os == 'macos-latest')
+        run: |
+          pip install "numpy<2.0" pillow
+          pytest -s
+
+      - name: Run Python tests with numpy<2.0 (ubuntu)
+        if: (matrix.python_version == '3.11' || matrix.python_version == '3.12' ) && (matrix.os == 'ubuntu-latest' || matrix.os == 'ubuntu-24.04')
         run: |
-          pip install "numpy<2.0" "pillow<10"
+          sudo apt-get install libjpeg-dev zlib1g-dev libpng-dev libtiff-dev libfreetype6-dev
```

Comment:
Could you add a comment on why this is needed?

Reply:
added